.github/workflows/wheel.yaml

```diff
@@ -1,102 +1,73 @@
-      - name: Check GCC version
-        run: |
-          # We want to build wheels on gcc 9 to be consistent with the torch
-          # binaries, which is why we're using ubuntu-20.04 and not
-          # ubuntu-latest. More details below about the GLIBCXX checks.
-          if ! gcc --version | grep -q -E "9\.[0-9]+\.[0-9]+"
-          then
-              echo "Wrong GCC version, exiting!"
-              gcc --version
-              exit 1
-          fi
-      - name: Build the torchcodec wheel
-        run: |
-          # Just for sanity, make sure FFmpeg isn't installed or needed for buidling.
-          .github/scripts/assert_ffmpeg_not_installed.sh
-
-          python -m pip install build
-          BUILD_AGAINST_ALL_FFMPEG_FROM_S3=1 python -m build . -vvv --no-isolation
-      - name: Validate wheel content
-        run: |
-          source .github/scripts/helpers.sh
-          wheel_path=$(pwd)/$(find dist -type f -name "*.whl")
-          echo "Wheel content:"
-          unzip -l $wheel_path
-
-          for ffmpeg_major_version in 4 5 6 7; do
-            assert_in_wheel $wheel_path torchcodec/libtorchcodec${ffmpeg_major_version}.so
-          done
-          assert_not_in_wheel $wheel_path libtorchcodec.so
-
-          for ffmpeg_so in libavcodec.so libavfilter.so libavformat.so libavutil.so libavdevice.so ; do
-            assert_not_in_wheel $wheel_path $ffmpeg_so
-          done
-
-          assert_not_in_wheel $wheel_path "^test"
-          assert_not_in_wheel $wheel_path "^doc"
-          assert_not_in_wheel $wheel_path "^benchmarks"
-
-          # See invoked python script below for details about this check.
-          extracted_wheel_dir=$(mktemp -d)
-          unzip -q $wheel_path -d $extracted_wheel_dir
-          symbols_matches=$(find $extracted_wheel_dir | grep ".so$" | xargs objdump --syms | grep GLIBCXX_3.4.)
-          python .github/scripts/check_glibcxx.py "$symbols_matches"
-      - uses: actions/upload-artifact@v4
-        with:
-          name: sdist-and-wheel-linux_x86_${{ matrix.python-version }}
-          path: dist/*
+    name: Build and Upload wheel
+    uses: pytorch/test-infra/.github/workflows/build_wheels_linux.yml@main
+    with:
+      repository: pytorch/torchcodec
+      ref: ""
+      test-infra-repository: pytorch/test-infra
+      test-infra-ref: main
+      build-matrix: ${{ needs.generate-matrix.outputs.matrix }}
+      post-script: packaging/post_build_script.sh
+      env-var-script: packaging/env_var_script.sh
+      smoke-test-script: packaging/fake_smoke_test.py
+      package-name: torchcodec
+      trigger-event: ${{ github.event_name }}
+      build-platform: "python-build-package"
+      wheel-build-extra-args: "-vvv --no-isolation"
 
   install-and-test:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.8', '3.12']
+        python-version: ['3.9']
```

Comment:
We move from 3.8-3.12 to just 3.9.

- We can't use 3.8 anymore because `test-infra` doesn't support it.
- on top of that, we can only use 3.9 here (for now) because by default, `test-infra` will only generate job on 3.9 on PRs. Generating job for all python versions can be done by labeling the PR with `ciflow/binaries/all` (I will eventually document this). What we *should* be doing here is letting this `install-and-test` job rely on the exact same python versions that were generated by the `generate-matrix` job. I haven't found a way to do that yet, but this should be doable, possibly requirement extensions to `test-infra`.


For now though, this is OK. We will still be able to checks all necessary python version when we push a new release.